client/components/tinymce/plugins/mentions/mentions.jsx

```diff
@@ -165,6 +190,68 @@ export class Mentions extends React.Component {
 		editor.getBody().focus();
 	}
 
+	onKeyDown = ( e ) => {
+		const { keyCode } = e;
+
+		if ( includes( [ VK.DOWN, VK.UP ], keyCode ) ) {
+			const selectedIndex = this.getSelectedSuggestionIndex();
+			let nextIndex;
+
+			if ( selectedIndex !== null ) {
+				if ( keyCode === VK.DOWN ) {
+					nextIndex = selectedIndex + 1;
+
+					if ( nextIndex === this.matchingSuggestions.length ) {
+						nextIndex = 0;
+					}
```

Comment:
here a bit of math can see us some `if`s

```js
nextIndex = ( selectedIndex + 1 ) % this.matchingSuggestions.length;
```

while maybe more than we need, this will wrap the selection around the edges as desired.